drafting.typ

```diff
@@ -199,38 +205,120 @@
+          margin.right = page.margin.right
+          margin.left = page.margin.left
+        } else if "inside" in page.margin.keys() {
+          margin.inside = page.margin.inside
+          margin.outside = page.margin.outside
+        }
+      } 
+
+      if (page.width, page.height) == (auto, auto) {
+        if ("right" in margin.keys() and auto in (margin.left, margin.right)) {
+          panic("If the page width *and* height are set to `auto`, neither left nor right margin" + " can be `auto`. Got (left, right) margin " + repr((
           margin.left,
           margin.right,
-        )) + ", and page (width, height) " + repr((page.width, page.height)))
+          )) + ", and page (width, height) " + repr((page.width, page.height)))
+        }
+        if ("inside" in margin.keys() and auto in (margin.inside, margin.outside)) {
+          panic("If the page width *and* height are set to `auto`, neither inside nor outside margin" + " can be `auto`. Got (inside, outside) margin " + repr((
+            margin.inside,
+            margin.outside,
+          )) + ", and page (width, height) " + repr((page.width, page.height)))
+        }
       }
       // https://github.com/typst/typst/issues/3636#issuecomment-1992541661
       let page-dims = (page.width, page.height).filter(x => x != auto)
+      let auto-size = 2.5 / 21 * calc.min(..page-dims)
+
       let defaults = (
         margin-left: margin.left,
         margin-right: margin.right,
+        margin-inside: margin.inside,
+        margin-outside: margin.outside,
       )
       for (k, v) in defaults {
         if v == auto {
-          defaults.at(k) = 2.5 / 21 * calc.min(..page-dims)
-        } else if type(v) == type(10pt - 100%) {
+          defaults.at(k) = auto-size
+        } else if type(v) == relative {
           defaults.at(k) = v.length
-        }
+        } // ignore if none
+      }
+      // NOTE: auto binding is dependent on text-direction which is not inferrable.
+      // Thus, we just assume ltr, unless bindis is set explicitly.
+      let binding-left = if page.binding == auto {
+          true
+      } else {
+        page.binding == left
       }
+
       set-margin-note-defaults(
         ..defaults,
         page-width: size.width,
+        page-binding-left: binding-left,
         ..kwargs,
       )
     })
   }
 }
 
+/// Show an outline of all notes
+///
+/// - title (string): Title of the outline
+/// - show-if-empty (bool): Whether to show the outline of there are no todos
+#let note-outline(title: "List of Todos", show-if-empty: false) = context {
+  show outline.entry.where(
+    level: 1
+  ): it => {
+    link(it.element.location())[
+      // box with color, followed by content
+      #it.body.children.first() #it.body.children.last()
+      #box(width: 1fr, repeat[.])
+      #it.page
+    ]
+  }
+  // only show if there actually are any todos
+  if show-if-empty or counter(figure.where(kind: "todo")).final().first() > 0 {
```

Comment:
Can we instead query for `<margin-note>` and/or `<inline-note>` labels instead of creating figures? Is there a distinct advantage of a hidden figure along with the labels? Should we replace label logic with the figure approach?

Reply:
I considered that, but to my knowledge, it would not be possible to create the boxes with the color of the todo in the outline if only using labels.

Reply:
Reading this again, we *could* query the labels, but the figure itself still has to persist as far as I can tell.